DirectEmployers/MyJobs
# [PD-2541] implement nuo wizard history

## What's this PR do?

Adds history handling to the NUO Process Record step. This allows for use of back, forward, and reload when navigating the entire app.

## Screenshots (if appropriate)

## Where should the reviewer start?

If you have no NUO records, run this script **be sure to replace the email field**
```
%autoindent
from datetime import datetime
from factory import fuzzy
from myjobs.models import User
from mypartners.models import (OutreachRecord, OutreachEmailAddress,
                               OutreachWorkflowState)

user = User.objects.get(email="_____ENTER_YOUR_EMAIL_HERE_____")
company = user.roles.filter(name="Admin").first().company
try:
    outreach_email = OutreachEmailAddress.objects.get(company=company,
                                                      email="detestemail")
except:
    outreach_email = OutreachEmailAddress(company=company,
                                          email="detestemail")
    outreach_email.save()

try:
    workflow = OutreachWorkflowState.objects.get(state="New")
except:
    workflow = OutreachWorkflowState(state="New")
    workflow.save()

i = 0
while i < 100:
    i += 1
    orr = OutreachRecord(date_added=datetime.now(),
                   outreach_email=outreach_email,
                   from_email=str(fuzzy.FuzzyText(length=6).fuzz()) + "@example.com",
                   email_body=fuzzy.FuzzyText(length=200).fuzz(),
                   current_workflow_state=workflow
                   )
    orr.save()
    print orr.pk
```

And if you don't have -access- to NUO, run thiiiss script.
```
from myjobs.models import User
from seo.models import BusinessUnit
import datetime

aa = AppAccess.objects.get(name="nuo")
user = User.objects.get(email="ENTER EMAIL HERE")
company = user.roles.filter(name="Admin").first().company
company.app_access.add(aa)
```

## How should this be manually tested?
Head to NUO. (https://secure.my.jobs/prm/view/nonuseroutreach/)
Click 'Review' on one of the records.
Start doing the process steps. Navigate back and forward occasionally.
Make sure it works.
Let me know if it doesn't.

## Any background context you want to provide?

This PR also has a couple of refactors in it.. The most notable one was swapping state.process.state to state.process.step (state as a variable name is pretty bad). And that refactor changed code in a loott of places. Sorry.
